extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -0,0 +1,97 @@
+const { usersBucket, mpuBucketPrefix } = require('arsenal').constants;
+const QueuePopulatorExtension =
+    require('../../lib/queuePopulator/QueuePopulatorExtension');
+const LIFECYCLE_BUCKETS_ZK_PATH = '/data/buckets';
+
+class LifecycleQueuePopulator extends QueuePopulatorExtension {
+
+    /**
+     * Pre-create the zookeeper path for bucket lifecycle nodes, if necessary.
+     * @param {Function} cb - The callback function.
+     * @return {undefined}
+     */
+    createZkPath(cb) {
+        const { zookeeperPath } = this.extConfig;
+        const path = `${zookeeperPath}${LIFECYCLE_BUCKETS_ZK_PATH}`;
+        return this.zkClient.getData(path, err => {
+            if (err) {
+                if (err.name !== 'NO_NODE') {
+                    this.log.error('Could not get zookeeper node path', {
+                        method: 'LifecycleQueuePopulator.createZkPath',
+                        error: err,
+                    });
+                    return cb(err);
+                }
+                return this.zkClient.mkdirp(path, err => {
+                    if (err) {
+                        this.log.error('Could not create path in zookeeper', {
+                            method: 'LifecycleQueuePopulator.createZkPath',
+                            zookeeperPath,
+                            error: err,
+                        });
+                        return cb(err);
+                    }
+                    return cb();
+                });
+            }
+            return cb();
+        });
+    }
+
+    /**
+     * Create a new zookeeper node for any bucket that has a lifecycle
+     * configuration. Remove the node if the bucket is deleted or the
+     * configuration is deleted.
+     * @param {Object} value - The record log entry value from metadata.
+     * @return {undefined}
+     */
+    _updateZkBucketNode(value) {
+        const { zookeeperPath } = this.extConfig;
+        const path = `${zookeeperPath}${LIFECYCLE_BUCKETS_ZK_PATH}/` +
+            `${value.owner}:${value.name}`;
+        // Remove existing node if deleting the bucket or its configuration.
+        if (value.deleted || value.lifecycleConfiguration === null) {
+            return this.zkClient.remove(path, err => {
+                if (err && err.name !== 'NO_NODE') {
```

Comment:
Can you add a descriptive info log here when the node has actually been removed (i.e not in NO_NODE case, which can still have a debug log)? It should not clutter the logs since it's only when bucket config changes, and can be useful info.

Reply:
Updated.